pkg/deploy/controller/deployment/controller.go

```diff
@@ -134,7 +135,7 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Also, it will scale down the failed deployment and scale back up
 		// the last successful completed deployment
 		if deployutil.IsDeploymentCancelled(deployment) {
-			deployerPods, err := c.podClient.getDeployerPodsFor(deployment.Namespace, deployment.Name)
+			deployerPods, err := deployutil.GetDeployerPodsFor(c.podClient, deployment)
```

Comment:
This is one small thing that bothers me for testability: if I'm glancing through this code, I don't know what this opaque call using my `client` is going to do, and so when developing a test, I now have to go diving through function calls to figure out what to mock. In this case it happens to be a single `List` call, but I've seen places where it rapidly turns into a call graph (with if/else branching and possibly OTHER function calls) that you can't easily decipher to know what you need to mock for a given case.

An alternative in this case would be to make another util function which returns a selector you can feed to List right here. There are other such functions already in the util to construct selectors. I think that's a decent compromise: shared selector creation, use the `List` client call directly to make it obvious what the dependencies are.

What do you think?